documentation/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,186 @@
+## Guide for migrating to `sdk/**/arm**` from `services/**/mgmt/**`
+
+This document is intended for users that are familiar with an older version of the Azure SDK For Go for management modules (`services/**/mgmt/**`) and wish to migrate their application to the next version of Azure resource management libraries (`sdk/**/arm**`)
+
+**For users new to the Azure SDK For Go for resource management modules, please see the [README for 'sdk/armcore`](https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/armcore) and the README for every individual package.**
+
+## Table of contents
+
+* [Prerequisites](#prerequisites)
+* [General Changes](#general-changes)
+    * [Authentication](#authentication)
+    * [Customized Policy](#customized-policy)
```

Comment:
I'd move the customized policy and custom HTTP client sections towards the end of the doc as these are advanced scenarios.

Reply:
Done